ledger/participant-integration-api/src/main/scala/platform/apiserver/error/ErrorInterceptor.scala

```diff
@@ -107,18 +109,40 @@ class ErrorListener[ReqT, RespT](delegate: ServerCall.Listener[ReqT], call: Serv
       // 2. We need to catch it and call `call.close` as otherwise gRPC will close the stream with a Status.UNKNOWN
       //    (see io.grpc.internal.ServerImpl.JumpToApplicationThreadServerStreamListener.internalClose)
       case t: StatusException =>
-        call.close(t.getStatus, t.getTrailers)
+        LogOnUnhandledFailureInClose(call.close(t.getStatus, t.getTrailers))
       case t: StatusRuntimeException =>
-        call.close(t.getStatus, t.getTrailers)
+        LogOnUnhandledFailureInClose(call.close(t.getStatus, t.getTrailers))
       case t: Throwable =>
         val e = LedgerApiErrors.InternalError
           .UnexpectedOrUnknownException(t = t)(
             new DamlContextualizedErrorLogger(logger, emptyLoggingContext, None)
           )
           .asGrpcError
-        call.close(e.getStatus, e.getTrailers)
+        LogOnUnhandledFailureInClose(call.close(e.getStatus, e.getTrailers))
     }
-
   }
+}
 
+private[error] object LogOnUnhandledFailureInClose {
```

Comment:
Maybe move this class (and the corresponding tests) to separate files?

Reply:
This class is only used here and has a specific error to these uses. I would make an exception here and leave it in here.